apps/src/lab2/views/components/Instructions/NavigationArea.tsx

```diff
@@ -112,28 +115,99 @@ const NavigationArea: React.FC<NavigationAreaProps> = ({
     }
   }, [validationMessage, isRunning]);
 
-  const canShowButton = useMemo(() => {
+  // Certain labs hide the continue button until the user has met the conditions for the next level. Example: Music lab.
+  const isLabHidesContinueButton = useMemo(() => {
+    return LABS_HIDE_CONTINUE_BUTTON_RESOURCE_PANEL.includes(appName);
+  }, [appName]);
```

Comment:
I'd really prefer to avoid lab-specific logic creeping into these shared components as much as possible :) can we pass this through as a prop? Something like `hideContinueIfDisabled = false` that Music Lab can set `true`?